.github/actions/l10n-update-pot/action.yml

```diff
@@ -0,0 +1,41 @@
+name: Update *.pot files
+
+# This action expects to have these repositories checked out:
+#   in directory `l10n` is the repository with weblate translations (e.g. dnf-l10n)
+#   in directory `source` is repository with sources (e.g. dnf)
+
+inputs:
+  branch:
+    required: true
+    description: Subdirectory where to store translations
+
+runs:
+  using: "composite"
+  steps:
+    - name: Generate *.pot files
+      shell: bash
+      run: |
+        # generate fresh *.pot file from source
+        pushd source > /dev/null
+          dnf --assumeyes build-dep *.spec
+          cmake .
+          make gettext-pot
+          POTFILES=$(git diff --name-only "*.pot")
+          if [ -z "$POTFILES" ]; then
+            exit 0
+          fi
+        popd > /dev/null
+
+        # copy new pot file to l10n repository
+        for POTFILE in ${POTFILES}; do
+          cp source/${POTFILE} l10n/${{inputs.branch}}/
+        done
+
+        # commit the changes
+        pushd l10n > /dev/null
+          git config --local user.email "packaging-team-maint@redhat.com"
+          git config --local user.name "Github action"
```

Comment:
I wouldn't use this name specific to GHA, what about "Automatic Translations Update" or something similar?